packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -139,17 +139,29 @@ export class RemoteChannelContext implements IChannelContext {
         // this as long as we support old attach messages
         if (attributes === undefined) {
             if (this.attachMessageType === undefined) {
-                throw new Error("Channel type not available");
+                throw new DataCorruptionError("Channel type not available", {
+                    channelId: this.id,
```

Comment:
We need to open follow-up issue to figure out what to do here regarding PII.
This data can have PII, and I believe we will have a class of cases like that where having this data is paramount for investigations, yet we can't collect PII. I believe the right solution here is to mark it properly such that telemetry layer is aware and can make determination of what to do. I believe should be able to collect it in staging and/or SPDF, but not submit in prod rings.

Note that this is related to containsPII flag that exits in our repo, that does not really work (it does not work because stack still contains data with potential PII, and also we attempt not to send it in internal rings, where we have to do it for similar reasons as here).

@markfields - FYI

Reply:
I see this is also tracking the work to filter PII from logging
https://github.com/microsoft/FluidFramework/issues/1920

Should I create a separate issue to track it for RemoteChannelContext or add a comment here linking to the existing one?

Reply:
Either way is fine

Reply:
Ok, but should this telemetry be currently sent to our logs then? To add the containsPII here though would strip this information from the logs needed for moving forward. Also, it would need to be added to the DataCorruptionError class itself, which may not always contain the PII content in the pendingStateManager error case.

Reply:
containsPII does not do anything useful today. And on top of that we need these IDs to make it actionable. We need proper end-to-end solution by Build only.

Reply:
Checked with @markfields and @vladsud  that this would only be an issue once 3P Fluid objects begin to be fetched so adding a comment for now to strip out this information before then